Core/include/Acts/Utilities/BinningData.hpp

```diff
@@ -211,6 +211,21 @@ class BinningData {
   BinningData() = default;
   ~BinningData() = default;
 
+  /// Equality operator
+  ///
+  /// @param bData is the binning data to be checked against
+  ///
+  /// @return a boolean indicating if they are the same
+  bool operator==(const BinningData& bData) const {
+    return (type == bData.type and option == bData.option and
+            binvalue == bData.binvalue and min == bData.min and
+            max == bData.max and step == bData.step and zdim == bData.zdim and
+            ((subBinningData == nullptr and bData.subBinningData == nullptr) ||
```

Comment:
```suggestion
            ((subBinningData == nullptr and bData.subBinningData == nullptr) or
```
if you're already using `and`?